services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java

```diff
@@ -432,6 +433,8 @@ public BrokeredIdentityContext getFederatedIdentity(String response) {
             }
 
             return identity;
+        } catch (IdentityBrokerUnmatchedEssentialClaimException e) {
```

Comment:
You would just re-throw `IdentityBrokerException` as per my first comment.

Reply:
Trying to carry the error message in the existing exception in the next commit